java-checks-test-sources/src/main/java/checks/AssignmentInSubExpressionCheck.java

```diff
@@ -128,4 +128,61 @@ void fun(List<Integer> list) {
       list.forEach(e -> field = field + e); // Compliant : ignore assignment expression in lambda
     }
   }
+
+  class java14 {
+    private static final java.util.Random RAND = new java.util.Random();
+    private int a;
+
+    // equivalent version of 'bar', which do not raise issues
+    void foo() {
+      String temp;
+      String temp2 = "";
+
+      switch (RAND.nextInt(10)) {
+        case 1 -> temp = "partial"; // Compliant
+        case 2 -> temp = "whole"; // Compliant
+        case 3, 5 -> temp = "none"; // Compliant
+        case 7 -> {
+          System.out.println(a = 0); // Noncompliant
+        }
+        default -> {
+          temp = "empty";
+          temp2 = "default";
+        }
+      }
+    }
+
+    void bar() {
+      String temp;
+      String temp2 = "";
+
+      switch (RAND.nextInt(5)) {
+        case 1:
+          temp = "partial";
+          break;
+        case 2:
+          temp = "whole";
+          break;
+        case 3, 5:
+          temp = "none";
+          break;
+        default:
+          temp = "empty";
+          temp2 = "default";
+          break;
+      }
+    }
+
+    private String b = "toto";
+    void qix() {
+      String s = switch (RAND.nextInt(10)) {
+        case 1 -> b = "yolo"; // Noncompliant
+        case 2 -> {
+          b = "tagada"; // Compliant
+          yield "turlutte";
+        }
+        default -> "boom";
+      };
```

Comment:
I may sound grumpy, but I think that we should avoid such string names and try to find ones describing what is happening for example (especially when some can have hidden meaning 😆)

Reply:
i agree with you. I actually forgot that I wrote this...